packages/cache/src/internal/requestUtils.ts

```diff
@@ -111,7 +111,9 @@ export async function retryTypedResponse<T>(
         return {
           statusCode: error.statusCode,
           result: null,
-          headers: {}
+          headers: {},
+          message: error.message,
+          typeKey: error.result.typeKey
```

Comment:
why do we need `typeKey`. can we not just use error message for logging an error?

Reply:
`typeKey` is the name of the exception thrown. This will help us narrow down error and take appropriate steps like 400 with multiple possible values of `typeKey` can be handled differently.

Reply:
Do we have a use case of handling different exception types differently? I can't think of such use case currently.